modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderMediator.java

```diff
@@ -122,11 +186,21 @@ public ValueProviderMediator(T containerModel, Supplier<MuleContext> muleContext
 
     ParameterModel parameterModel = parameters.get(0);
 
-    ValueProviderFactoryModelProperty factoryModelProperty =
-        parameterModel.getModelProperty(ValueProviderFactoryModelProperty.class)
-            .orElseThrow(() -> new ValueResolvingException(format("The parameter with name '%s' is not an Values Provider",
-                                                                  parameterName),
-                                                           INVALID_VALUE_RESOLVER_NAME));
+    ValueProviderFactoryModelProperty factoryModelProperty;
+    if (targetPath != null) {
+      factoryModelProperty =
+          parameterModel.getModelProperty(FieldsValueProviderFactoryModelProperty.class)
+              .map(fieldsValueProvider -> fieldsValueProvider.getFieldsValueProviderFactories().get(targetPath))
+              .orElseThrow(() -> new ValueResolvingException(format("The parameter with name '%s' is not an Values Provider",
```

Comment:
The parameter with name --- does not have a Value Provider associated with the targetPath ---